spec/latest/core/core.jsonld

```diff
@@ -249,7 +252,7 @@
       "label": "returns",
       "comment": "The information returned by the Web API on success",
       "domain": "hydra:Operation",
-      "range": "hydra:Class",
+      "range": "hydra:Resource",

```

Comment:
Hmm, beforehand `:op <hydra:returns> <schema:Person>` meant, that the Operation `:op` returns resources of the class `schema:Person`.

No I would understand the same triple like: The Operation returns the resource `schema:Person` itself, instead of resources of that type.

Same applies to hydra:expects, the other way arround.

Reply:
`Class` is itself subClass of `Resource`. Hence, nothing really changes in principle.

I think it makes sense, so that one would be able to declare that an operation returns either an RDF instance of a `Class` or a non-RDF representation.

Reply:
The question is, what does `hydra:returns` *mean*? It should have one meaning shouldn't it?

The current meaning as I understand it is "returns instances of the mentioned hydra:Class"

What is the meaning, if I do not name a hydra:Class as the object of `hydra:returns`?

You say the use case is non-RDF representations. How would hat look like, and what does `hydra:returns` mean in that case? What if I just state `:op hydra:returns </some/resource>.`. Does the operation always return `</some/resource>`?

Reply:
You won't be able to express the fact that the operation returns a resource matching some spefication (being of a given type/class in that case) in the pure RDF. This should be written in the specification.

By saying `hydra:returns` `some non-RDF resource specification` I'd say the operation returns resources that match that specification (i.e. have media type of the one specified).